mongodbatlas/resource_mongodbatlas_project_api_key.go

```diff
@@ -143,13 +225,53 @@ func resourceMongoDBAtlasProjectAPIKeyUpdate(ctx context.Context, d *schema.Reso
 
 	if d.HasChange("role_names") {
 		updateRequest.Roles = expandStringList(d.Get("role_names").(*schema.Set).List())
-
-		_, err := conn.ProjectAPIKeys.Assign(ctx, projectID, apiKeyID, updateRequest)
-		if err != nil {
-			return diag.FromErr(fmt.Errorf("error updating API key: %s", err))
+		if updateRequest.Roles != nil {
+			_, err := conn.ProjectAPIKeys.Assign(ctx, projectID, apiKeyID, updateRequest)
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("error updating API key: %s", err))
+			}
 		}
 	}
 
+	if d.HasChange("project_assignment") {
+		// get the current api_keys and the new api_keys with changes
```

Comment:
Super minor, but just for consistency in the comments, these could all start with capitalized verbs
```suggestion
		// Getting the current api_keys and the new api_keys with changes
```